src/components/NewCardForm.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import "../styles/NewCardForm.css";
+
+const kDefaultFormState = {
+  message: "",
+};
+
+const NewCardForm = ({ onCardSubmit }) => {
+  const [formFields, setFormFields] = useState(kDefaultFormState);
+
+  const handleChange = (event) => {
+    const fieldName = event.target.name;
+    const fieldValue = event.target.value;
+
+    const newFormFields = { ...formFields, [fieldName]: fieldValue };
+    setFormFields(newFormFields);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    onCardSubmit(formFields);
+
+    setFormFields(kDefaultFormState);
+  };
+
+  return (
+    <div className="new-card-form">
+      <form className="form-content" onSubmit={handleSubmit}>
+        <h2>Create New Card</h2>
+        <div>
+          <label htmlFor="message">Message: </label>
+          <input
+            type="text"
+            name="message"
+            value={formFields.message}
+            onChange={handleChange}
+          ></input>
+        </div>
+
+        <div>
+          <input
+            className="submit-button"
+            type="submit"
+            value="Add card!"
+          ></input>
+        </div>
+      </form>
+    </div>
```

Comment:
Nice use of semantic tags overall! There are a couple places, such as here and the new board form, where I would consider if something like `section` or `article` might be a better fit for the enclosing tag.